core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -0,0 +1,135 @@
+ * continues, but the exit code is 16, instead of 0. Disabling an already disabled plugin does nothing. It only restart
+ * if, at least, one plugin has been disabled and the restart option was set.
+ *
+ * @since 2.136
+ */
+@Extension
+public class DisablePluginCommand extends CLICommand {
+
+    @Argument(metaVar = "plugins", required = true, usage = "Plugins to be disabled.")
+    private List<String> pluginNames;
+
+    @Option(name = "-restart", usage = "Restart Jenkins after disabling plugins.")
+    private boolean restart;
+
+    @Override
+    public String getShortDescription() {
+        return Messages.DisablePluginCommand_ShortDescription();
+    }
+
+    @Override
+    protected void printUsageSummary(PrintStream stderr) {
+        stderr.println(Messages.DisablePluginCommand_PrintUsageSummary());
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        jenkins.checkPermission(Jenkins.ADMINISTER);
+
+        PluginManager manager = jenkins.getPluginManager();
+        int result = 0;
+        boolean somePluginDisabled = false; // to ensure we restart when it needs to
+        for (String pluginName : pluginNames) {
+            switch (disablePlugin(manager, pluginName)) {
+                case DISABLED:
+                    if(!somePluginDisabled) somePluginDisabled = true;
+                    break;
+                case NOT_DISABLED_DEPENDANTS:
+                    if(result != 16) result = 16; // Custom error code to indicate one plugin, at least, can't be disabled
+            }
+        }
+
+        if (restart && somePluginDisabled) {
+            jenkins.safeRestart();
+        }
+
+        return result;
+    }
+
+    /**
+     * Try to disable a  plugin.
+     * @param manager The PluginManager.
+     * @param shortName The name of the plugin to disable.
+     * @return The result of the disabling of this plugin. See {@link DISABLING_STATUS}
+     * @throws IOException An exception disabling the plugin. See {@link PluginWrapper#disable()}
+     */
+    private DISABLING_STATUS disablePlugin(PluginManager manager, String shortName) throws IOException {
+        PluginWrapper plugin = manager.getPlugin(shortName);
+        if (plugin == null) {
+            throw new IllegalArgumentException(Messages.DisablePluginCommand_NoSuchPlugin(shortName)); // exit with 3
+        }
+
+        if (!plugin.isEnabled()) {
+            stdout.format(Messages.DisablePluginCommand_Already_Disabled(shortName));
+            stdout.println();
+            return DISABLING_STATUS.ALREADY_DISABLED;
+        }
+
+        Set<String> dependants = plugin.getDependants();
+        for (String dependant : dependants) {
+            PluginWrapper dependantPlugin = manager.getPlugin(dependant);
+            if (dependantPlugin != null && dependantPlugin.isEnabled()) {
+                // TO-DO: stdout or stderr? the process continues but as the result code is not 0, it looks like stderr
```

Comment:
Please use `TODO` instead of `TO-DO` for better support in IDEs.

Reply:
I'll remove the comment, It was there to point it out to all reviewers